pkg/metrics/builder/builder_test.go

```diff
@@ -0,0 +1,241 @@
+package builder
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+func getDescStruct(name string, desc string, constLabels map[string]string, variableLabels []string) string {
+	lpStrings := make([]string, 0, len(constLabels))
+	for key, value := range constLabels {
+		lpStrings = append(lpStrings, fmt.Sprintf("%s=%q", key, value))
+	}
+	sort.Strings(lpStrings)
+
+	return fmt.Sprintf("Desc{fqName: %q, help: %q, constLabels: {%s}, variableLabels: %v}",
+		name, desc, strings.Join(lpStrings, ","), variableLabels)
+}
+
+func TestMetricBuilderHistogram(t *testing.T) {
+	labels := []string{"test1", "test2"}
+	labelValues := map[string]string{"test1": "test1", "test2": "test2"}
```

Comment:
can we use something useful so that this test helps users understand the usage of the api.. tests can act as documentation and exmaples...